### DISCUSSION TITLE
Why the refusal to add a bottom nav bar?

### DISCUSSION THREAD

**Discussion by Fmoment1** [bromite/bromite#1685] [Ideas]
This would be a huge win for accessibility. I'm frankly puzzled at how so many issues have been rejected regarding this. Unless I'm wrong, this doesn't compromise privacy in any way. So why not add it? 

There are also other UI/UX related features worth considering like adding a separate share, copy and edit buttons alongside the address bar, and recent websites below it. Like this example from Brave browser: 
![Screenshot_20211229-124238](https://user-images.githubusercontent.com/28984045/147636744-cca6d487-fbb8-446c-8210-96d1403f2faa.png)

Please consider adding these! It would be a great way forward towards modernizing the browser, which is already pretty good otherwise. 

Here's the bottom bar example: 
![Screenshot_20211229-124809](https://user-images.githubusercontent.com/28984045/147636938-daabe7cb-e79e-4255-9a0c-1d5d539a151f.png)

**Comment by uazo** [comment]
> Why the refusal to add a bottom nav bar?

I think it is because it **could** be very complex to maintain.

but @csagan5 would you accept a patch that does this?

![1](https://user-images.githubusercontent.com/29201891/148518311-474043af-8dba-41be-bc24-5219c9e857a3.gif)

**Reply by uazo** [reply]
@csagan5 the final patch if you want to take a look, tested with android 6, 9, 10 and 11, even in tablet mode and with different dpi screens and orientations.
I realize it's beyond bromite's purpose and I have to tell the truth, I didn't think I was capable of doing it, but in the end it was just a matter of tweaking a few offsets here and there.
at most I learned how the chromium compositor works :)

https://uazo.github.io/superpatch/bromite/commit/ce445b9779453029acbb93cf2ce4bf996e53d163

**Reply by uazo** [reply]
@csagan5 an answer on this too?

**Reply by csagan5** [reply]
@uazo can you provide a link to actual code please? But generally: no, it is unrelated to privacy.

**Reply by uazo** [reply]
sure, [lastest version](https://github.com/uazo/bromite/blob/master-v97/build/patches/Move-toptoolbar-to-bottom.patch) for v97 (but I developed it in v96) and [here](https://github.com/uazo/bromite-buildtools/releases) you can find it compiled for arm64 (hash you see matches the commit, all build process is open and visible in actions)

the latter is bromite without branding (+ all the fixes that would close the various issues) but without `Automated-domain-substitution.patch` (I hate search and replace in the code and above all it breaks the downloads of the chromium tools in my build script).

> But generally: no, it is unrelated to privacy.

for the record, it's been a month since I've been using it on my device, and I find it convenient...

**Reply by uazo** [reply]
ah, one more thing, if it can help to change your position.
the patch works at a low level in the code, ie above the ui and all chromium functions, ie in general it should touch code that hardly changes over time (obviously I don't know for sure).
I also found some bugs and notified the chromium team ([1281195](https://bugs.chromium.org/p/chromium/issues/detail?id=1281195) [1285311](https://bugs.chromium.org/p/chromium/issues/detail?id=1285311))

**Reply by csagan5** [reply]
This seems a big patch, it will probably break from release to release; I am hesitant to add further non-privacy-related patches that increase the maintenance burden.

**Reply by uazo** [reply]
> non-privacy-related patches that increase the maintenance burden.

this is not true and you know it. maintenance is just as much as any other, unless we just change flags.
it gets easier if we create the right tools, I'm working on it but I'm waiting for your ideas. at what point have you arrived?

> it will probably break from release to release

bringing it to 98 there was only one method of which the signature changed, it does not seem demanding

**Reply by csagan5** [reply]
> this is not true and you know it

I am surprised of the accusation of lying; regardless, this patch is not privacy-related and you know it.

> it gets easier if we create the right tools, I'm working on it but I'm waiting for your ideas. at what point have you arrived?

I will release tools for better builds and patching when they are more mature.

**Reply by uazo** [reply]
> I am surprised of the accusation of lying;

if I appeared offensive, I apologize.

**Reply by csagan5** [reply]
> > I am surprised of the accusation of lying;
> 
> if I appeared offensive, I apologize.

We have been over this: before implementing anything we need to discuss it. If you prefer to implement it anyways, use it in your own builds, please do not expect it to be always included.

**Reply by uazo** [reply]
> We have been over this: before implementing anything we need to discuss it.

as I write, I didn't think I was capable of it

> please do not expect it to be always included.

never thought about it

> I am hesitant to add further non-privacy-related patches that increase the maintenance burden.

hope you can change your mind

**Reply by csagan5** [reply]
Patches unrelated to privacy must be kept to a minimum; we have discussed this elsewhere, but only option I see is to include it as an optional patch under `contrib/`.

**Comment by KjmT6459** [comment]
I would really like this too. When FF first introduced it, I instantly turned it off. But after giving it another try, I'm kicking myself for ignoring it, and I really don't want to go back to a top-bar. (I'm still using bromite due to FF's shortcomings on Android)

My phone is under 6in, and I have to painfully stretch to reach the top-bar. Using a friend's new bigger phone, I exclusively have to use two hands to reach the top-bar. This is absolutely an accessibility problem and I wholeheartedly support it's inclusion.

**Reply by bat999** [reply]
> ...When FF first introduced it, I instantly turned it off. But after giving it another try, I'm kicking myself for ignoring it, and I really don't want to go back to a top-bar...

Hi
It's available in Fennec F-Droid too.
I suppose the feature is maintained upstream by Mozilla.

"* Customizable navigation bar position"

**Comment by starbuckbok** [comment]
Just checking in to offer my wholehearted support for including the option to move the toolbar to the bottom. I hope the dev will consider it.

**Comment by NePoCz** [comment]
Not sure if that would anyhow decrease the maintanance burden, but Brave also has the bottom toolbar, so you could get some help or inspiration there? 

For myself, the missing bottom toolbar is the bigest downside of Bromite on larger phones (i.e. most of the newer ones), but also understand that too many features can cause maintanance trouble. Whatever the decision, thanks for a great browser

**Reply by uazo** [reply]
> Brave also has the bottom toolbar, so you could get some help or inspiration there?

before starting, I checked the code of brave and also that of kiwi, their logic is different and in some aspects more complex to maintain. they started with revert the chromium commit which duplicates some views, in my opinion my patch is bigger but simpler.

**Comment by coxackie** [comment]
so, I saw that a patch has been submitted to enable bottom toolbar, but no instructions how to do it. Is there a flag we need to enable? I thought it would be in the accessibility menu...

**Reply by csagan5** [reply]
The feature is not mentioned in the CHANGELOG nor in the README because it is not used yet.
